### PR TITLE
Get main type and subtype from oncotree.mskcc.org

### PR DESCRIPTION
**Pull request by victoria34**

### CODE REVIEW

File: src/main/webapp/app/service/trial.service.ts [hunk 1]
```diff
@@ -96,23 +97,37 @@ export class TrialService {
...
+                    'query': item,
+                    'version': this.oncotreeVersion,
+                    'type': 'mainType'
+                });
+            }
+            // prepare subtypes by maintype
+            const queries =  {
+                'queries': mainTypeQueries
+            };
+            this.connectionService.getSubType(queries).subscribe((response: Array<any>) => {
```
Comment (zhx828) [inline src/main/webapp/app/service/trial.service.ts]: When the API failed, we should give a message saying either queries invalid or the service is not available depends on the http code.

File: src/main/webapp/app/service/trial.service.ts [hunk 2]
```diff
@@ -342,4 +362,20 @@ export class TrialService {
         }
         return result;
     }
+    getErrorResponse(error: HttpErrorResponse, type: string) {
+        if (error.status === 400) {
+            alert('Sorry, your query is invalid.');
```
Comment (zhx828) [inline src/main/webapp/app/service/trial.service.ts]: Don't we use `dialogs.notify`?

File: src/main/webapp/app/service/trial.service.ts [hunk 3]
```diff
@@ -342,4 +362,20 @@ export class TrialService {
...
+        } else if (error.status === 404) {
+            alert('Sorry, we cannot find ' + type + '.');
+        } else if (error.status === 503) {
+            alert('Sorry, required data source is unavailable now.');
+        } else {
+            alert('Sorry, unexpected error happens. Our development team has been notified.');
```
Comment (zhx828) [inline src/main/webapp/app/service/trial.service.ts]: The message needs to be used in the callback of sendEmail, or at least after sending email.

File: src/main/webapp/app/service/trial.service.ts (hunk 2 above)
Comment (victoria34) [reply src/main/webapp/app/service/trial.service.ts]: No. I can replace alert to dialogs.notify in next PR.

Comment (zhx828) [reply src/main/webapp/app/service/trial.service.ts]: Oh, the dialog is used in oncokb curate not in MM? If it is an easy change, could you add that in?

Comment (victoria34) [reply src/main/webapp/app/service/trial.service.ts]: Angular Dialog Service is a complete Angularjs service. I may need to spend time looking for a Angular 2+ service.

**Pull request merged**